test/integration/controllermanager/shoot/maintenance/maintenance_test.go

```diff
@@ -319,7 +339,14 @@ var _ = Describe("Shoot Maintenance controller tests", func() {
 		It("Do not update Shoot machine image in maintenance time: AutoUpdate.MachineImageVersion == false && expirationDate does not apply", func() {
 			Expect(kutil.SetAnnotationAndUpdate(ctx, testClient, shoot, v1beta1constants.GardenerOperation, v1beta1constants.ShootOperationMaintain)).To(Succeed())
 
-			Expect(waitForExpectedMachineImageMaintenance(ctx, log, testClient, shoot, testMachineImage, false, time.Now().Add(time.Second*10))).To(Succeed())
+			Consistently(func(g Gomega) gardencorev1beta1.ShootMachineImage {
+				g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(shoot), shoot)).To(Succeed())
+				return *shoot.Spec.Provider.Workers[0].Machine.Image
+			}).Should(Equal(gardencorev1beta1.ShootMachineImage{Name: testMachineImage.Name, Version: testMachineImage.Version}))
+			Consistently(func(g Gomega) gardencorev1beta1.ShootMachineImage {
```

Comment:
The first `Consistently` will block until the image is actually updated so my suggestion here would be to combine both `Consistently` invocations into one. Maybe something like
```
Consistently(func(g Gomega) {
	g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(shoot), shoot)).To(Succeed())
	g.Expect(*shoot.Spec.Provider.Workers[0].Machine.Image).To(Equal(gardencorev1beta1.ShootMachineImage{Name: testMachineImage.Name, Version: testMachineImage.Version}))
	g.Expect(*shoot.Spec.Provider.Workers[1].Machine.Image).To(Equal(gardencorev1beta1.ShootMachineImage{Name: testMachineImage.Name, Version: testMachineImage.Version}))
}).Should(Succeed())
```

I would do the same for the `Eventually` invocations below, since the machine images are expected to be updated in one reconciliation right?